server/server.go

```diff
@@ -1960,6 +1971,22 @@ func (s *Server) AcceptLoop(clr chan struct{}) {
 	clr = nil
 }
 
+// InProcessConn returns an in-process connection to the server,
+// avoiding the need to use a TCP listener for local connectivity
+// within the same process. This can be used regardless of the
+// state of the DontListen option.
+func (s *Server) InProcessConn() (net.Conn, error) {
+	pl, pr := net.Pipe()
+	if !s.startGoRoutine(func() {
+		s.createClient(pl)
```

Comment:
It should be this:
```
s.createClient(pl)
s.grWG.Done()
```
I mentioned that the call of `s.grWG.Done()` when startGoRoutine() failed needed to be removed, but failed to mention that it has to be done once the function started in startGoRoutine() finishes.

Reply:
Have added this in.

Reply:
I don't see this in the commit that says you did (just the move of `end := ..`

Reply:
Oh, what. VS Code apparently decided it wanted to block on the language server and not save changes, so that's incredibly helpful.